File: event-bus-rabbit/src/main/dotnet/amqp/ActiveSubscription.cs
using System;

namespace pegasus.eventbus.amqp
{
	/// <summary>
	/// Represents a subscription with an active queue listener, pulling messages from the Bus.
	/// </summary>
	public class ActiveSubscription
	{
		private string _queueName;
		private bool _isDurable;
		private QueueListener _listener;
		private bool _isActive;
		
		
		public string QueueName
		{
			get { return _queueName; }
		}
		
		public bool IsForDurableQueue
		{
			get { return _isDurable; }
		}
		
		public QueueListener Listener
		{
			get { return _listener; }
		}
		
		public bool IsActive
		{
			get { return _isActive; }
			set { _isActive = value; }
		}
		
		
		public ActiveSubscription(string queueName, bool queueIsDurable, QueueListener listener)
		{
			_queueName = queueName;
			_isDurable = queueIsDurable;
			_listener = listener;
		}
	}
}

File: event-bus/event-bus-client/src/main/java/gov/ment/eventbus/amqp/ActiveSubscription.java
package gov.ment.eventbus.amqp;

/**
 * Represents a Subscription with an active queue listener, pulling messages
 * from the Bus.
 * 
 * @author Ken Baltrinic (Berico Technologies)
 */
class ActiveSubscription {

  private final String queueName;
  private final Boolean queueIsDurable;
  final QueueListener listener;
  private boolean isActive = true;

  /**
   * Instantiate the class supplying the queue information and listener
   * 
   * @param queueName
   *          Name of the Queue being watched
   * @param queueIsDurable
   *          Is the Queue durable?
   * @param listener
   *          The listener watching for events
   */
  public ActiveSubscription(String queueName, Boolean queueIsDurable, QueueListener listener) {

    this.queueName = queueName;
    this.queueIsDurable = queueIsDurable;
    this.listener = listener;
  }

  /**
   * Get the Queue Name
   * 
   * @return Queue Name
   */
  public String getQueueName() {
    return queueName;
  }

  /**
   * Is the Queue Durable?
   * 
   * @return true if it is durable
   */
  public Boolean getQueueIsDurable() {
    return queueIsDurable;
  }

  /**
   * Get the Listener watching the Queue
   * 
   * @return Listener
   */
  public QueueListener getListener() {
    return listener;
  }

  /**
   * Is the subscription currently Active?
   * 
   * @return true if active.
   */
  public boolean isActive() {
    return isActive;
  }

  /**
   * Toggle whether the queue is active
   * 
   * @param queueIsDeleted
   *          Has the queue been deleted?
   */
  public void setIsActive(boolean queueIsDeleted) {
    this.isActive = queueIsDeleted;
  }

}
